test/client_test.cc

```diff
@@ -1,3 +1,16 @@
+// This test relies on fork(). Somehow the integration test server excepts when running
+// under TSAN because of Envoy not being able to find its configuration file with die_after_fork
+// disabled. The code below seems to suggest that TSAN doesn't support this scenario.
+// https://github.com/llvm-mirror/compiler-rt/blob/master/lib/tsan/rtl/tsan_interceptors.cc#L999
+// Further root cause analysis is not worth it at this point, because this code will be deprecated
+// soon in favor of either https://github.com/envoyproxy/nighthawk/pull/60 or moving this end-to-end
+// testing to python.
+// Thus we just disable this test when we detect we are running under TSAN.
+#ifndef __has_feature
+#define __has_feature(x) 0 // Compatibility with non-clang compilers.
```

Comment:
Can you use the `ENVOY_CONFIG_TSAN` defined in the `.bazelrc` instead? This will a bit more robust and easier to handle during Google import. Thanks.

Reply:
Done; but hmm, Envoy has a spot that might need a change for consistency [here](https://github.com/envoyproxy/envoy/blob/32e4d286668731594eb5c81ed664bd144d8d2d88/test/exe/main_common_test.cc#L119).

Reply:
This worked, but broke after merging master. Reverting to use `__has_feature(...` made it work again, but it may be a good idea to figure out why `ENVOY_CONFIG_TSAN` no longer is defined.